java/src/main/native/ai_onnxruntime_OrtSession.c

```diff
@@ -13,227 +13,286 @@
  * Method:    getNumOutputs
  * Signature: (JJ)J
  */
-JNIEXPORT jlong JNICALL Java_ai_onnxruntime_OrtSession_getNumOutputs
-  (JNIEnv * jniEnv, jobject jobj, jlong apiHandle, jlong handle) {
-    (void) jobj; // Required JNI parameter not needed by functions which don't need to access their host object.
-    const OrtApi* api = (const OrtApi*) apiHandle;
-    size_t numOutputs;
-    checkOrtStatus(jniEnv,api,api->SessionGetOutputCount((OrtSession*)handle, &numOutputs));
-    return numOutputs;
+JNIEXPORT jlong JNICALL Java_ai_onnxruntime_OrtSession_getNumOutputs(JNIEnv* jniEnv, jobject jobj, jlong apiHandle, jlong handle) {
+  (void)jobj;  // Required JNI parameter not needed by functions which don't need to access their host object.
+  const OrtApi* api = (const OrtApi*)apiHandle;
+  size_t numOutputs = 0;
+  checkOrtStatus(jniEnv, api, api->SessionGetOutputCount((OrtSession*)handle, &numOutputs));
+  return numOutputs;
 }
 
 /*
  * Class:     ai_onnxruntime_OrtSession
  * Method:    getOutputNames
  * Signature: (JJJ)[Ljava/lang/String;
  */
-JNIEXPORT jobjectArray JNICALL Java_ai_onnxruntime_OrtSession_getOutputNames
-  (JNIEnv * jniEnv, jobject jobj, jlong apiHandle, jlong sessionHandle, jlong allocatorHandle) {
-    (void) jobj; // Required JNI parameter not needed by functions which don't need to access their host object.
-    const OrtApi* api = (const OrtApi*) apiHandle;
-    OrtAllocator* allocator = (OrtAllocator*) allocatorHandle;
-
-    // Setup
-    char *stringClassName = "java/lang/String";
-    jclass stringClazz = (*jniEnv)->FindClass(jniEnv, stringClassName);
-
-    // Get the number of outputs
-    size_t numOutputs = Java_ai_onnxruntime_OrtSession_getNumOutputs(jniEnv, jobj, apiHandle, sessionHandle);
-
-    // Allocate the return array
-    jobjectArray array = (*jniEnv)->NewObjectArray(jniEnv,safecast_size_t_to_jsize(numOutputs),stringClazz, NULL);
-    for (uint32_t i = 0; i < numOutputs; i++) {
-        // Read out the output name and convert it to a java.lang.String
-        char* outputName;
-        checkOrtStatus(jniEnv,api,api->SessionGetOutputName((OrtSession*)sessionHandle, i, allocator, &outputName));
-        jstring name = (*jniEnv)->NewStringUTF(jniEnv,outputName);
-        (*jniEnv)->SetObjectArrayElement(jniEnv, array, i, name);
-        checkOrtStatus(jniEnv,api,api->AllocatorFree(allocator,outputName));
+JNIEXPORT jobjectArray JNICALL Java_ai_onnxruntime_OrtSession_getOutputNames(JNIEnv* jniEnv, jobject jobj, jlong apiHandle, jlong sessionHandle, jlong allocatorHandle) {
+  (void)jobj;  // Required JNI parameter not needed by functions which don't need to access their host object.
+  const OrtApi* api = (const OrtApi*)apiHandle;
+  OrtSession* session = (OrtSession*)sessionHandle;
+  OrtAllocator* allocator = (OrtAllocator*)allocatorHandle;
+
+  // Setup
+  char* stringClassName = "java/lang/String";
+  jclass stringClazz = (*jniEnv)->FindClass(jniEnv, stringClassName);
```

Comment:
Same comments as for inputs

Reply:
I'll modify the char* to be static, but as mentioned above I'm not sure we can cache the class pointer.

Reply:
We can probably make this string global static and share among the functions. I will let the class go.

Reply:
Ok.